core/store/src/lib.rs

```diff
@@ -184,10 +184,15 @@ impl StoreUpdate {
         self.transaction.update_refcount(column, key, value)
     }
 
+    // Saves a given value.
+    // Must not be used for RC columns - please use 'update refcount' instead.
     pub fn set(&mut self, column: DBCol, key: &[u8], value: &[u8]) {
+        debug_assert!(!column.is_rc());
```

Comment:
So this would actually break what I know think might be correct usage in recompress_storage. It now reads data using iter_without_rc_logic and writes it to a new empty database using set. As I understand things now, it’s a bit hacky but correct way to do it.

Reply:
yeah, that's quite hacky.. 

how about we add a new iterator that returns (key, value, refcount) - and then we could use 'update_refcount' call rather than set?